src/mapping/nav2_map_server/include/nav2_map_server/base_map_loader.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_MAP_SERVER__BASE_MAP_LOADER_HPP_
+#define NAV2_MAP_SERVER__BASE_MAP_LOADER_HPP_
+
+#include <libgen.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <fstream>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+
+class BaseMapLoader
+{
+public:
+  std::string fname;
```

Comment:
Do these need to be public? Also, do they need to be in the abstract base class. I'd rather see a set of virtual methods and a virtual destructor (only).